client-java/controller-api/pom.xml

```diff
@@ -17,6 +17,11 @@
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-engine</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>bson</artifactId>
+            <version>${mongodb.version}</version>
+        </dependency>
```

Comment:
unfortunately we can't have this. we cannot have third-party libraries in the client-java that are not shaded. and it is a further issue when this libraries are used by the SUT (it ll be a mess). an option would be to use "provided", but we need to discuss its ramifications before doing something like that

Reply:
After your changes to the third-party replacement mechanisms, do I still need to remove this dependency?